src/components/Navbar/interfaces.tsx

```diff
@@ -22,3 +24,24 @@ export interface NavLink extends NavLinkElement {
   /** An array to hold a dropdown Links */
   children: NavLinkElement[]
 }
+
+export interface Search {
+  /** Defines the placeholder text. */
+  placeholderText?: string | 'Search'
+  /** Defines the button text. */
+  buttonText?: string | 'Search'
+  /** Defines the button variant. */
+  buttonColor?:
+    | 'primary'
+    | 'secondary'
+    | 'success'
+    | 'warning'
+    | 'danger'
+    | 'info'
+    | 'light'
+    | 'dark'
+  /** Handles the on click search button event */
+  onClickButton?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+  /** Handles the on change search form event */
+  onChangeInput?: (event: React.ChangeEvent<HTMLInputElement>) => void
```

Comment:
I think that these two fields must be required. If they are no provided, then the search would not have any functionality.